coopbri/knit
# Project restructure

## Description

This PR represents the first major work after forking https://github.com/wclr/yalc.

- Added PR template
- Added Husky precommit hook with lint-staged
- Added CI (Changesets, test Actions workflow, release Actions workflow)
- Replaced tslint (deprecated) with eslint for linting
- Added Prettier for code formatting
- Added Renovate config
- Upgraded dependencies
- Switched test runner to Jest (with SWC engine)
- Updated README documentation
- Moved command files into `src/commands`
- Added tsup for bundling

## Test Steps

- Verify tests pass locally
- Verify Action workflows are successful and not flaky
- Verify build works locally (`yarn build`)
- Verify built binary works locally after building (`yarn knit ...`)
